File: BE-FindingJob/src/main/java/com/example/befindingjob/service/CompanyService.java
package com.example.befindingjob.service;

import com.example.befindingjob.dto.CompanyDTO;
import com.example.befindingjob.entity.Company;
import com.example.befindingjob.model.ApiResponse;

import java.util.List;

public interface CompanyService {
    ApiResponse<Void> addCompany(CompanyDTO companyDTO);
    ApiResponse<Void> updateCompany(CompanyDTO companyDTO);
    ApiResponse<Void> deleteCompany(int companyId);
    ApiResponse<List<Company>> getAllCompaniesByUserId(String token);
}

File: JobHub/app/src/main/java/com/example/jobhub/service/CompanyService.kt
package com.example.jobhub.service

import com.example.jobhub.dto.CompanyDTO
import com.example.jobhub.entity.Company
import com.example.jobhub.model.ApiResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Header
import retrofit2.http.POST

interface CompanyService {
    @POST("api/company/add-company")
    fun addCompany(@Body companyDTO: CompanyDTO): Call<ApiResponse<Void>>

    @POST("api/company/update-company")
    fun updateCompany(@Body companyDTO: CompanyDTO): Call<ApiResponse<Void>>

    @POST("api/company/delete-company")
    fun deleteCompany(@Body companyId: Int): Call<ApiResponse<Void>>

    @POST("api/company/get-all-companies-by-userId")
    fun getAllCompaniesByUserId(@Header("token") token: String): Call<ApiResponse<List<Company>>>
}